src/Pos/Block/Worker.hs

```diff
@@ -186,17 +189,23 @@ verifyCreatedBlock blk =
         }
 
 #if defined(WITH_WALLET)
--- | This one just triggers every @max (slotDur / 4) 5@ seconds and
--- asks for current tip. Does nothing when recovery is enabled.
-behindNatWorker :: (WorkMode ssc m, SscWorkersClass ssc) => (WorkerSpec m, OutSpecs)
-behindNatWorker = worker requestTipOuts $ \sendActions -> do
+-- | When we're behind NAT, other nodes can't send data to us and thus we
+-- won't get blocks that are broadcast through the network – we have to reach
+-- out to other nodes by ourselves.
```

Comment:
I've been thinking about this and found that time-warp sessions would be really useful https://github.com/serokell/time-warp-nt/pull/24 . The node behind NAT can start (and hold open for as long as possible) a session with the addressable node, allowing the addressable node to push to the unaddressable one. This way it actually makes sense for the unaddressable node to define listeners.